### DISCUSSION TITLE
blog/new-angular-coding-style

### DISCUSSION THREAD

**Discussion by giscus** [angular-experts-io/angular-experts-website-comments#21] [General]
# blog/new-angular-coding-style

Embracing the New Angular: A Fresh Perspective on how to write Angular code.

https://angularexperts.io/blog/new-angular-coding-style?ref=dailydev

**Comment by jeanfwl** [comment]
Thanks for the blog; I appreciate the knowledge!

I have a question: Do you use readonly in your variable declarations, most for service injection?

I've made it a habit to always include private readonly for my services or any variables that don’t change. Do you think it’s really necessary, particularly for service injection?

**Reply by Dozorengel** [reply]
I have the same question, please, anybody respond

**Comment by SrdjanMilic** [comment]
What a great article! Thanks for sharing it.

**Reply by Haze272** [reply]
Nivek, you have a broken link to your GitHub's account

**Reply by Haze272** [reply]
mistaken reply

**Comment by Haze272** [comment]
Nivek, you have a broken link to your GitHub's account

**Comment by radekbusa** [comment]
When we were migrating some more complex components involving ngOnChanges with checks related to previousValue and currentValue which were later executing side effects, such as writing to backend user settings storage, we had some pretty tough time reimplementing that to Signals, the only thing we came up with was using an effect and assigning previous value to a separate private signal. So in current state of affairs, I wouldn't say that computed or effect (easily) replaces ngOnChanges.
However, the transition from private to ECMAScript private fields inspired me a bunch and thank you for that 😎

**Reply by szheleshchenko** [reply]
You should get in touch with [Jean Meche](https://github.com/JeanMeche), one of the Angular team members. In one of his talks, he asked people to share use cases where getting rid of standard lifecycle hooks is not suitable or convenient.

**Comment by Sp94dev** [comment]
Dude! Assigning effects to a variable changed my life! That’s brilliant because it allows us to extract "React-like hooks" as pure functions and reuse them!
